File: DailyChallenges/CPP/LeetCode_232.cpp
class MyQueue {
private:
    queue<int> q;
public:
    MyQueue() {
    }
    
    void push(int x) {
        q.push(x);
    }
    
    int pop() {
        int x = q.front(); q.pop();
        return x;
    }
    
    int peek() {
        return q.front();
    }
    
    bool empty() {
        return q.empty();
    }
};

File: DailyChallenges/Scala/LeetCode_232.scala
class MyQueue() {
    var l = List.empty[Int]

    def push(x: Int) {
        l = x :: l
    }

    def pop(): Int = {
        val x = l.last
        l = l.init
        x
    }

    def peek(): Int = {
        l.last
    }

    def empty(): Boolean = {
        l.isEmpty
    }

}
